File: py/src/ai/h2o/sparkling/ml/models/H2OMOJOSettings.py
#
# Licensed to the Apache Software Foundation (ASF) under one or more
# contributor license agreements.  See the NOTICE file distributed with
# this work for additional information regarding copyright ownership.
# The ASF licenses this file to You under the Apache License, Version 2.0
# (the "License"); you may not use this file except in compliance with
# the License.  You may obtain a copy of the License at
#
#    http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

from h2o.utils.typechecks import assert_is_type
from pyspark.ml.param import *
from pyspark.ml.wrapper import JavaWrapper


class H2OMOJOSettings(JavaWrapper):

    def __init__(self,
                 predictionCol="prediction",
                 detailedPredictionCol="detailed_prediction",
                 withDetailedPredictionCol=False,
                 convertUnknownCategoricalLevelsToNa=False,
                 convertInvalidNumbersToNa=False,
                 namedMojoOutputColumns=True):
        self._java_obj = None

        assert_is_type(predictionCol, str)
        assert_is_type(detailedPredictionCol, str)
        assert_is_type(withDetailedPredictionCol, bool)
        assert_is_type(convertUnknownCategoricalLevelsToNa, bool)
        assert_is_type(convertInvalidNumbersToNa, bool)
        assert_is_type(namedMojoOutputColumns, bool)
        self.predictionCol = predictionCol
        self.detailedPredictionCol = detailedPredictionCol
        self.withDetailedPredictionCol = withDetailedPredictionCol
        self.convertUnknownCategoricalLevelsToNa = convertUnknownCategoricalLevelsToNa
        self.convertInvalidNumbersToNa = convertInvalidNumbersToNa
        self.namedMojoOutputColumns = namedMojoOutputColumns

    def toJavaObject(self):
        self._java_obj = self._new_java_obj("ai.h2o.sparkling.ml.models.H2OMOJOSettings",
                                            self.predictionCol,
                                            self.detailedPredictionCol,
                                            self.withDetailedPredictionCol,
                                            self.convertUnknownCategoricalLevelsToNa,
                                            self.convertInvalidNumbersToNa,
                                            self.namedMojoOutputColumns)
        return self._java_obj

    @staticmethod
    def default():
        return H2OMOJOSettings()

File: r/src/R/ai/h2o/sparkling/ml/models/H2OMOJOSettings.R
H2OMOJOSettings.default <- function() {
  H2OMOJOSettings()
}

#' @export H2OMOJOSettings
H2OMOJOSettings <- setRefClass("H2OMOJOSettings",
                               fields = list(predictionCol = "character",
                                             detailedPredictionCol = "character",
                                             convertUnknownCategoricalLevelsToNa = "logical",
                                             convertInvalidNumbersToNa = "logical",
                                             withContributions = "logical",
                                             withInternalContributions = "logical",
                                             withPredictionInterval = "logical",
                                             withLeafNodeAssignments = "logical",
                                             withStageResults = "logical",
                                             dataFrameSerializer = "character",
                                             scoringBulkSize = "integer"),
                               methods = list(
                                 initialize = function(predictionCol = "prediction",
                                                       detailedPredictionCol = "detailed_prediction",
                                                       convertUnknownCategoricalLevelsToNa = FALSE,
                                                       convertInvalidNumbersToNa = FALSE,
                                                       withContributions = FALSE,
                                                       withInternalContributions = FALSE,
                                                       withPredictionInterval = FALSE,
                                                       withLeafNodeAssignments = FALSE,
                                                       withStageResults = FALSE,
                                                       dataFrameSerializer = "ai.h2o.sparkling.utils.JSONDataFrameSerializer",
                                                       scoringBulkSize = 1000L) {
                                   .self$predictionCol <- predictionCol
                                   .self$detailedPredictionCol <- detailedPredictionCol
                                   .self$convertUnknownCategoricalLevelsToNa <- convertUnknownCategoricalLevelsToNa
                                   .self$convertInvalidNumbersToNa <- convertInvalidNumbersToNa
                                   .self$withContributions <- withContributions
                                   .self$withInternalContributions <- withInternalContributions
                                   .self$withPredictionInterval <- withPredictionInterval
                                   .self$withLeafNodeAssignments <- withLeafNodeAssignments
                                   .self$withStageResults <- withStageResults
                                   .self$dataFrameSerializer <- dataFrameSerializer
                                   .self$scoringBulkSize <- scoringBulkSize
                                 },
                                 toJavaObject = function() {
                                   sc <- spark_connection_find()[[1]]
                                   invoke_new(sc, "ai.h2o.sparkling.ml.models.H2OMOJOSettings",
                                              .self$predictionCol,
                                              .self$detailedPredictionCol,
                                              .self$convertUnknownCategoricalLevelsToNa,
                                              .self$convertInvalidNumbersToNa,
                                              .self$withContributions,
                                              .self$withInternalContributions,
                                              .self$withPredictionInterval,
                                              .self$withLeafNodeAssignments,
                                              .self$withStageResults,
                                              .self$dataFrameSerializer,
                                              .self$scoringBulkSize)
                                 }
                               ))
